charts/k8ssandra-cluster/templates/cleaner/batch_job.yaml

```diff
@@ -0,0 +1,31 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ .Release.Name }}-cleaner-job-k8ssandra
+  labels:
+{{ include "k8ssandra-cluster.labels" . | indent 4 }}
+  annotations:
+    "helm.sh/hook": pre-delete
+    "helm.sh/hook-delete-policy": hook-succeeded,before-hook-creation
+spec:
+  backoffLimit: 3
+  template:
+    metadata:
+      labels:
+{{ include "k8ssandra-cluster.labels" . | indent 8 }}
+    spec:
+      restartPolicy: Never
+      serviceAccountName: {{ .Release.Name }}-cleaner-k8ssandra
+      containers:
+        - name: cleaner-job-k8ssandra
+          image: {{ .Values.cleaner.image }}
+          imagePullPolicy: IfNotPresent
+          env:
+            - name: POD_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
+          args:
```

Comment:
I had to make a couple changes to `args` in order to get the cleaner to run. I changed it to the following:

```yaml
          args:
            - -clean
            - --release={{ .Release.Name }}
```

If I leave the `k8ssandra-client` arg, the `clean` flag evaluates to false.

I think that the `flag` package requires requires use of `=` instead of whitespace to separate the flag name and value. With whitespace, I get a parsing error.

Reply:
No, flag package supports multiple methods (-flag, -flag=x, -flag x, --flag etc), and that should work:

```
michael@nina ~/p/g/d/k/c/k8ssandra-client (deletion_job)> go build .
michael@nina ~/p/g/d/k/c/k8ssandra-client (deletion_job) [2]> POD_NAMESPACE=default ./k8ssandra-client -clean --release seemeee
2021/01/05 16:28:50 Cleaning resources for uninstall
2021/01/05 16:28:50 Removing CassandraDatacenter(s) managed in release seemeee from namespace default
michael@nina ~/p/g/d/k/c/k8ssandra-client (deletion_job)> 
```

Reply:
Here's with the new patch, from kubernetes logs:

```
2021-01-05 17:19:17 | 2021-01-05T15:19:15.730094791Z stderr F 2021/01/05 15:19:15 Cleaning resources for uninstall
-- | --
  |   | 2021-01-05 17:19:17 | 2021-01-05T15:19:16.28315523Z stderr F 2021/01/05 15:19:16 Removing CassandraDatacenter(s) managed in release demo-cluster from namespace default
```